src/Type/UnionType.php

```diff
@@ -45,6 +51,12 @@ class UnionType implements CompoundType
 
 	use NonGeneralizableTypeTrait;
 
+	public const EQUAL_UNION_CLASSES = [
+		DateTimeInterface::class => [DateTimeImmutable::class, DateTime::class],
+		Throwable::class => [Error::class, Exception::class],
+		Traversable::class => [IteratorAggregate::class, Iterator::class],
+	];
```

Comment:
Not sure if I need to put this constant somewhere else (like UnionTypeHelper) / as a static function or with a better naming.

I support any number of classes in values, just in case.